### PR TITLE
Resolve table function based on StorageEngine provided function resolver

### PR DESCRIPTION
**Pull request by penghuo**
### Description
1. Resolve table function based on StorageEngine provided function resolver.
2. The JDBC datasource PR based on this PR. 
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1331
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java [hunk 1]
```diff
@@ -8,5 +8,5 @@
 public enum DataSourceType {
   PROMETHEUS,
   OPENSEARCH,
-  FILESYSTEM
+  JDBC
```
Comment (penghuo) [inline core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java:11]: FILESYSTEM supported has been removed in https://github.com/opensearch-project/sql/pull/1284.

File: core/build.gradle [hunk 2]
```diff
@@ -62,7 +62,7 @@ dependencies {
 test {
     useJUnitPlatform()
     testLogging {
-        events "passed", "skipped", "failed"
+        events "skipped", "failed"
```
Comment (penghuo) [inline core/build.gradle:65]: Ignore log PASSED test cases.

**Review by anirudha — APPROVED** [review]

**Comment by penghuo** [comment]
To reviewer: Ignore failed bwc test.

**Review by YANG-DB — DISMISSED** [review]

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by YANG-DB — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Pull request merged**